heracal/omni.py

```diff
@@ -532,6 +565,102 @@ def from_fits(filename, pols=None, bls=None, ants=None, verbose=False):
                             
     return meta, gains, v, x
 
+def write_uvdata_vis(filename, aa, m, v, xtalk=False, returnuv=True):
+    '''
+    Given meta information and visibilities, write out a uvfits file.
+    filename: filename to write uvfits object.(str)
+    aa      : aipy antenna array object (object)
+    m       : dictionary of information (dict)
+    v       : dictionary of visibilities with keys antenna pair and pol (dict)
+    xtalk   : visibilities given are xtalk visibilities. (bool)
+    returnuv: return uvdata object. If returned, file is not written. (bool)
+    '''
+
+    pols = v.keys()
+    bls = v[pols[0]].keys()
+
+    uv = UVData()
+    if xtalk:
+        uv.Ntimes = 1
+    else:
+        uv.Ntimes = len(m['times'])
+    uv.Npols = len(pols)
+    uv.Nbls = len(v[pols[0]].keys())
+    uv.Nblts = uv.Nbls * uv.Ntimes
+    uv.Nfreqs = len(m['freqs'])
+    data = {}
+    for p in v.keys():
+        if p not in data.keys():
+            data[p] = []
+        for bl in v[p].keys():
+            data[p].append(v[p][bl])
+        data[p] = np.array(data[p]).reshape(uv.Nblts, uv.Nfreqs)
```

Comment:
I think the `for` loop above can be simplified significantly. Does the first `if` condition ever evaluate to False? — I don't think so, because otherwise the `data[p].append()` call won't work because `data[p]` ends up being a Numpy array, which doesn't support that.

I think you might just be able to write something like:

```
for p, bl_dict in v.items():
    data[p] = np.array(bl_dict.values()).reshape((uv.Nblts, uv.Nfreqs))
```